c_wrapper/lua-mongoc-collection.c

```diff
@@ -730,15 +744,33 @@ lua_mongo_collection_drop_index(lua_State *L)
 {
     bool ret = false;
     bool throw_error = false;
+    bool idxTable = false;
 
     collection_t *collection;
     bson_error_t error;
 
-    const char *index_name;
+    char *index_name = "";
     int absolute_luaBSONObjects_index = 2;
     int index_name_index = 3;
 
-    index_name = luaL_checkstring(L, index_name_index);
+    if(lua_isstring(L, index_name_index)){
+        index_name = luaL_checkstring(L, index_name_index);
+    }else if(lua_istable(L, index_name_index)){
+        idxTable = true;
+        lua_pushvalue(L, index_name_index);
+        lua_pushnil(L);
+        while(lua_next(L, -2)){
```

Comment:
You repeat code here. You can probably make a new method to do what you need to do. However, if we use `mongoc_collection_keys_to_index_string` it is probably small enough not to break everything out into another method.